apps/server/src/modules/provisioning/strategy/oidc/service/schulconnex-user-provisioning.service.ts

```diff
@@ -70,4 +75,18 @@ export class SchulconnexUserProvisioningService {
 
 		return savedUser;
 	}
+
+	private async checkUniqueEmail(
+		email: string,
+		systemId: string,
+		schoolId?: string,
+		externalId?: string
+	): Promise<void> {
+		const foundUsers: UserDO[] = await this.userService.findByEmail(email);
```

Comment:
I would like to see this check in the user service because its more a user related thing to check existing emails.
Just call this function here and throw